File: Coding-Interviews/32-II-从上到下打印二叉树II.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
#include <queue>
#include <vector>
using namespace std;
class Solution
{
public:
    vector<vector<int>> levelOrder(TreeNode* root)
    {
        if (!root)
            return {};
        vector<vector<int>> ret;
        queue<TreeNode*>    q;
        q.push(root);
        while (!q.empty()) {
            vector<int> cur;
            auto        cnt = q.size();
            for (int i = 0; i < cnt; i++) {
                auto front = q.front();
                q.pop();
                cur.push_back(front->val);
                if (front->left)
                    q.push(front->left);
                if (front->right)
                    q.push(front->right);
            }
            ret.push_back(cur);
        }
        return ret;
    }
};

File: Coding-Interviews/32-II-从上到下打印二叉树II.py
# Definition for a binary tree node.
import collections
from typing import List


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        q = collections.deque()
        q.append(root)
        res = []
        while q:
            cnt = len(q)
            line = []
            for _ in range(0, cnt):
                node = q.popleft()
                line.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            res.append(line)
        return res
